diff --git a/node_modules/react-quill/lib/component.js b/node_modules/react-quill/lib/component.js
index 9e578a3..525f2fc 100644
--- a/node_modules/react-quill/lib/component.js
+++ b/node_modules/react-quill/lib/component.js
@@ -216,7 +216,7 @@
 
   /*
    * The component gained or lost focus.
-   * TODO: quill.hasFocus() is not public API, this should be replaced.
+   * FIXME: quill.hasFocus() is not public API, this should be replaced.
    */
   _onFocus(range) {
     if (this.props.onFocus) {
@@ -231,7 +231,7 @@
 
   /*
    * The component gained or lost focus.
-   * TODO: quill.hasFocus() is not public API, this should be replaced.
+   * FIXME: quill.hasFocus() is not public API, this should be replaced.
    */
   _onBlur(previousRange) {
     if (this.props.onBlur) {
@@ -404,7 +404,11 @@
     }
     if (this.editor) {
       this.setEditorSelection(this.editor, range);
-      this.editor.focus();
+      // Fix for 'findDOMNode is deprecated in StrictMode'
+      // Check if this.editor and this.editor.root exist before focusing
+      if (this.editor.root) {
+        this.editor.focus();
+      }
     }
   }
 
