diff --git a/node_modules/react-quill/lib/index.js b/node_modules/react-quill/lib/index.js
index 994a86f..c0a78b5 100644
--- a/node_modules/react-quill/lib/index.js
+++ b/node_modules/react-quill/lib/index.js
@@ -165,7 +165,8 @@
 	},
 
 	getEditingArea: function(){
-		return ReactDOM.findDOMNode(this.editingArea);
+		// N.B. this is a temporary fix for findDOMNode deprecation in React 18.
+		return this.editingArea.current;
 	},
 
 	getEditor: function(){
diff --git a/node_modules/react-quill/lib/component.js b/node_modules/react-quill/lib/component.js
index c8d29c8..14407b7 100644
--- a/node_modules/react-quill/lib/component.js
+++ b/node_modules/react-quill/lib/component.js
@@ -482,7 +482,7 @@
 	},
 
 	getEditingArea: function(){
-		return ReactDOM.findDOMNode(this.editingArea);
+		return this.editingArea.current;
 	},
 
 	/*
