diff --git a/node_modules/react-quill/lib/index.js b/node_modules/react-quill/lib/index.js
index 4704af4..285e786 100644
--- a/node_modules/react-quill/lib/index.js
+++ b/node_modules/react-quill/lib/index.js
@@ -1,6 +1,6 @@
 'use strict';
 
-var React = require('react');
+var React = require('react');
 var ReactDOM = require('react-dom');
 var Quill = require('quill');
 var PropTypes = require('prop-types');
@@ -522,7 +522,7 @@
 	},
 
 	getEditingArea: function(){
-		return ReactDOM.findDOMNode(this.editingArea);
+		return this.editingArea;
 	},
 
 	getEditor: function(){
@@ -822,7 +822,7 @@
 			classNames.push('ql-container');
 			classNames.push('ql-snow');
 			return React.createElement('div', {
-					ref: function(el) { this.editingArea = el; }.bind(this),
+					ref: function(el) { this.editingArea = el; }.bind(this),
 					className: classNames.join(' '),
 					style: this.props.style
 				});